aws/resource_aws_appstream_stack_test.go

```diff
@@ -0,0 +1,346 @@
+					testAccCheckAwsAppStreamStackExists(resourceName, &stackOutput),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_time"),
+					naming.TestCheckResourceAttrNameGenerated(resourceName, "name"),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+					resource.TestCheckResourceAttr(resourceName, "description", descriptionUpdated),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsAppStreamStack_withTags(t *testing.T) {
+	var stackOutput appstream.Stack
+	resourceName := "aws_appstream_stack.test"
+	description := "Description of a test"
+	descriptionUpdated := "Updated Description of a test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckAwsAppStreamStackDestroy,
+		ErrorCheck:        testAccErrorCheck(t, appstream.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsAppStreamStackConfigComplete(description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppStreamStackExists(resourceName, &stackOutput),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_time"),
+					naming.TestCheckResourceAttrNameGenerated(resourceName, "name"),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+					resource.TestCheckResourceAttr(resourceName, "description", description),
+				),
+			},
+			{
+				Config: testAccAwsAppStreamStackConfigWithTags(descriptionUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppStreamStackExists(resourceName, &stackOutput),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_time"),
+					naming.TestCheckResourceAttrNameGenerated(resourceName, "name"),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+					resource.TestCheckResourceAttr(resourceName, "description", descriptionUpdated),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key", "value"),
+					resource.TestCheckResourceAttr(resourceName, "tags_all.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags_all.Key", "value"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAppStreamStackExists(resourceName string, appStreamStack *appstream.Stack) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).appstreamconn
+		resp, err := conn.DescribeStacks(&appstream.DescribeStacksInput{Names: []*string{aws.String(rs.Primary.ID)}})
+
+		if err != nil {
+			return err
```

Comment:
For 3x bonus points, add some message to the error, usually what you're doing: "Checking stack exists by describing stacks: %w"